commit cda9459da736481c6ff7e4368c994b98533a28cc
Author: Borislav Petkov <bp@suse.de>
Date:   Wed Jun 10 16:20:56 2015 +0200

    EDAC, mce_amd_inj: Set MISCV on injection
    
    When during injection we populate MCi_MISC by writing into misc, we need
    to set the MiscV bit in the corresponding MCi_STATUS register which
    denotes that there's valid info in the MCi_MISC register.
    
    Signed-off-by: Borislav Petkov <bp@suse.de>

diff --git a/drivers/edac/mce_amd_inj.c b/drivers/edac/mce_amd_inj.c
index b92986b03f93..4c73e4d03d46 100644
--- a/drivers/edac/mce_amd_inj.c
+++ b/drivers/edac/mce_amd_inj.c
@@ -191,6 +191,9 @@ static void do_inject(void)
 	unsigned int cpu = i_mce.extcpu;
 	u8 b = i_mce.bank;
 
+	if (i_mce.misc)
+		i_mce.status |= MCI_STATUS_MISCV;
+
 	if (inj_type == SW_INJ) {
 		amd_decode_mce(NULL, 0, &i_mce);
 		return;